openpype/plugins/publish/validate_unique_subsets.py

```diff
@@ -49,17 +53,41 @@ def process(self, context):
                                  "{}".format(instance.name))
                 continue
 
-            instance_per_asset_subset[(asset, subset)].append(instance)
+            # for cases where multiple versions under same subset are published
+            version = instance.data.get("version")
+            if version is None:
+                # this might happen when workfile version sync is disabled
+                # we still need to validate cases where multiple "next version"
+                # instances are published under same subset
+                self.log.info("Instance found without `version` data: "
+                              "{}".format(instance.name))
+            else:
+                # check if asset and subset combination is unique
+                # just to inform the user with warning about the case
+                if any([
+                    asset == a and subset == s
+                    for a, s, _ in instance_per_asset_subset.keys()
+                ]):
+                    self.log.warning(
+                        "Instance found with non unique `asset` and "
+                        "`subset` data: {}".format(instance.name)
+                    )
+
+            key = asset, subset, version
+            instance_per_asset_subset[key].append(instance)
 
         non_unique = []
-        for (asset, subset), instances in instance_per_asset_subset.items():
-
+        for key, instances in instance_per_asset_subset.items():
+            asset, subset, version = key
```

Comment:
Likely you're better off doing something like:
```python
instances_per_asset_subset[asset, subset].append(instance)
```

And then:
```python
for (asset, subset), instances in instances_per_asset_subset.items():
    
    if len(instances) < 2:
        # All good, ignore
        continue

    # An instance *may* explicitly set a specific version to publish into
    # and if so, it is allowed to publish into the same subset if the versions
    # differ between the instances. So we detect whether the instances
    # are set to publish to a different version
    instances_by_version = defaultdict(list)
    for instance in instances:
        version = instance.data.get("version")
        instances_by_version[version].append(instance)

    multiple_instances_per_version = False
    for version, version_instances in instances_by_version.items():
        if len(version_instances) > 2:
            multiple_instances_per_version = True
            non_unique.append(
                "{asset} > {subset} > {version}".format(
                    asset=asset, subset=subset, version=version
                )
            )

    if not multiple_instances_per_version:
        versions = ", ".join(str(version) for version in instances_by_version)
        self.log.warning(
            "Multiple instances are writing to '%s' > '%s' but to different versions. "
            "This is allowed but is considered an edge case. "
            "The versions to be published are: %s",
            asset, subset, versions
        )
   
```

Only issue with this currently is that it _also_ logs it if the same instance will error out after due to not having unique versions.